shared-module/displayio/Palette.h

```diff
@@ -52,6 +53,20 @@ typedef struct {
     bool transparent; // This may have additional bits added later for blending.
 } _displayio_color_t;
 
+typedef struct {
+    uint32_t pixel;
+    uint16_t x;
+    uint16_t y;
+    uint8_t tile;
+    uint16_t tile_x;
+    uint16_t tile_y;
+} displayio_input_pixel_t;
+
+typedef struct {
+    uint32_t pixel;
+    bool opaque;
+} displayio_output_pixel_t;
```

Comment:
Why introduce these two structs? The output contains no new data and the input could be params instead.

Reply:
The displayio_output_pixel_t opaque was added to remove ambiguity with the return value, so it clearly defines the intent. The display_io_input_pixel_t was added to reduce the number of input arguments into the functions to make for cleaner code, and reduce the amount of data copied onto the stack.

Reply:
Ok, let's remove the return value completely if we want to replace it with the output struct. Otherwise we risk them getting out of sync.